src/shared/bg_pmove.cpp

```diff
@@ -3603,9 +3598,65 @@ PM_Weapon
 Generates weapon events and modifies the weapon counter
 ==============
 */
+
+// for weapons with long, noticeable cooldown (mantis pounce)
+static void cooldown( pml_t & pml )
```

Comment:
This could be a single line instead of a separate function. `weaponCharge = std::min(0, weaponCharge - pml.msec)`

Reply:
Part of the intent there is to have weapons properties as isolated as possible, so that one day it would be possible to no longer have per-weapon code.
But right, maybe it's too soon for that kind of things.